cachi2/core/package_managers/npm.py

```diff
@@ -585,30 +625,68 @@ def _update_package_json_files(
     return package_json_projectfiles
 
 
+def _merge_npm_sbom_properties(components: list[NpmComponentInfo]) -> list[NpmComponentInfo]:
+    """Deduplicate the NpmComponentInfo objects and merge their dev and bundled properties."""
+    merged_components: dict[str, NpmComponentInfo] = {}
+    for component in components:
+        purl = component["purl"]
+        if purl not in merged_components:
+            merged_components[purl] = component
+            continue
+
+        # The dev and bundled properties for a package are True only if True for all
+        # components with the same purl
+        merged_component = merged_components[purl]
+        merged_component["bundled"] = component.get("bundled", False) and merged_component.get(
+            "bundled", False
+        )
+        merged_component["dev"] = component.get("dev", False) and merged_component.get("dev", False)
+
+    return list(merged_components.values())
+
+
+def _generate_component_list(component_info: list[NpmComponentInfo]) -> list[Component]:
+    """Convert a list of NpmComponentInfo objects into a list of Component objects for the SBOM."""
+    merged_component_info = _merge_npm_sbom_properties(component_info)
+    properties = {
+        "dev": Property(name="cdx:npm:package:development", value="true"),
+        "bundled": Property(name="cdx:npm:package:bundled", value="true"),
+    }
+    components: list[Component] = []
+    for merged_component in merged_component_info:
+        component = Component.from_package_dict(merged_component)
```

Comment:
The idea behind adding this `from_package_dict` function was to simplify the code when we were still in the copy-paste-from-cachito phase. The function is too generic, and the `package_dict` used to refer to Golang packages. 

I think it's better to avoid using it, and eventually refactor it out of the code.